eole/modules/multi_headed_attn.py

```diff
@@ -460,12 +460,27 @@ def _compute_attention(
                         query,
                         key,
                         value,
-                        ~mask if mask is not None else None,
+                        ~mask if (mask is not None and not causal) else None,
                         self.dropout_p,
                         is_causal=causal,
                     )
             attn = None
-
+        elif (
```

Comment:
why can't we use the same code above ?

Reply:
We definately could in the current case, I was splitting it out so we could add further MPS specifics later.

Reply:
For now I would prefer to keep the code as compact as possible until you add further MPS stuff later on. Thanks.